crates/revm/src/optimism/handler_register.rs

```diff
@@ -18,25 +18,34 @@ use revm_precompile::{secp256r1, PrecompileSpecId};
 use std::string::ToString;
 use std::sync::Arc;
 
-pub fn optimism_handle_register<DB: Database, EXT>(handler: &mut EvmHandler<'_, EXT, DB>) {
-    spec_to_generic!(handler.cfg.spec_id, {
-        // validate environment
-        handler.validation.env = Arc::new(validate_env::<SPEC, DB>);
-        // Validate transaction against state.
-        handler.validation.tx_against_state = Arc::new(validate_tx_against_state::<SPEC, EXT, DB>);
-        // Load additional precompiles for the given chain spec.
-        handler.pre_execution.load_precompiles = Arc::new(load_precompiles::<SPEC, EXT, DB>);
-        // load l1 data
-        handler.pre_execution.load_accounts = Arc::new(load_accounts::<SPEC, EXT, DB>);
-        // An estimated batch cost is charged from the caller and added to L1 Fee Vault.
-        handler.pre_execution.deduct_caller = Arc::new(deduct_caller::<SPEC, EXT, DB>);
-        // Refund is calculated differently then mainnet.
-        handler.execution.last_frame_return = Arc::new(last_frame_return::<SPEC, EXT, DB>);
-        handler.post_execution.reward_beneficiary = Arc::new(reward_beneficiary::<SPEC, EXT, DB>);
-        // In case of halt of deposit transaction return Error.
-        handler.post_execution.output = Arc::new(output::<SPEC, EXT, DB>);
-        handler.post_execution.end = Arc::new(end::<SPEC, EXT, DB>);
-    });
+pub fn optimism_handle_register<DB: Database, EXT>(
+    with_reward_beneficiary: bool,
+) -> HandleRegisters<EXT, DB> {
+    HandleRegisters::Box(Box::new(move |handler: &mut EvmHandler<'_, EXT, DB>| {
+        spec_to_generic!(handler.cfg.spec_id, {
+            // validate environment
+            handler.validation.env = Arc::new(validate_env::<SPEC, DB>);
```

Comment:
For performance we should downgrade these `Arc`s into `Box`es as well.

Reply:
this should be done on `main` branch instead, right?

Reply:
let's move this discussion to #2 if needed